src/schema/models/schemaManager.ts

```diff
@@ -87,66 +84,69 @@ export class SchemaManager {
       if (schema.enableExternalFetch) {
         //check each tag if it's an explode field and put it in explodeKeys
         if (schema.explodeKeysSet.has(key) && value !== null) {
-          explodeKeys.push(value.toString());
-        }
-
-        //check each tag if it's a domain field and put it in domainFieldsKeys
-        if (domainFields.has(key.toUpperCase()) && value !== null) {
-          domainFieldsKeys.push(`${key.toUpperCase()}:${value.toString()}`);
+          explodeKeys.push(`${keyConstructor(this.explodePrefix, key, value.toString())}`);
+        } else if (value !== null) {
+          domainKeys.push(`${keyConstructor(this.domainPrefix, key.toUpperCase(), value.toString())}`);
         }
       }
 
       return { ...acc, [key]: value };
     }, {});
 
-    let mappedDomainFields = {};
-    let explodeFieldsKeys = {};
+    let domainFieldsTags = {};
+    let explodeFieldsTags = {};
 
-    if (domainFieldsKeys.length > 0) {
-      mappedDomainFields = await this.getDomainFieldsCodedValues(domainFieldsKeys);
+    if (domainKeys.length > 0) {
+      domainFieldsTags = await this.getDomainFieldsCodedValues(domainKeys);
     }
 
     if (explodeKeys.length > 0) {
-      explodeFieldsKeys = await this.getExplodeFields(explodeKeys);
+      explodeFieldsTags = await this.getExplodeFields(explodeKeys);
     }
 
-    finalTagsObj = { ...finalTagsObj, ...mappedDomainFields, ...explodeFieldsKeys };
+    finalTags = { ...finalTags, ...domainFieldsTags, ...explodeFieldsTags };
 
     if (this.schemas[name].addSchemaPrefix) {
-      //for each key add a system name prefix
-      finalTagsObj = Object.entries(finalTagsObj).reduce((acc, [key, value]) => {
+      // for each key add a system name prefix
+      finalTags = Object.entries(finalTags).reduce((acc, [key, value]) => {
```

Comment:
you can use map + https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries
not sure whats better